feat(input): implement group type input, enter-to-submit and auto-clear on submission

flamintune opened the pull request:
### Introduction

As a newcomer to this project and making my first pull request, I want to express my eagerness to contribute effectively. If there's anything in my approach that seems off or inadvertently steps over any lines, I kindly ask for your understanding and welcome any constructive feedback.

### Changes Made

Refactored the Tab Group Type creation process to be more efficient by enabling users to complete the action through a text input field followed by pressing the Enter key. This update minimizes the need for mouse clicks and streamlines the user workflow.
![UI](https://github.com/MichaelYuhe/ai-group-tabs/assets/79519446/9a37ae7e-da75-44de-bdd2-15ba45e6062d)

### Closing Remarks

As this is my first contribution, I am open to learning and improving. Should there be any oversight or misstep on my part, I'd appreciate your guidance to help me grow as a contributor. Thank you for your patience and understanding.

MichaelYuhe commented on src/popup.tsx:
```diff
@@ -60,6 +58,29 @@ const Popup = () => {
+          onSubmit={(e) => {
+            setNewType('');
+            setTypes([...types, newType]);
+            e.preventDefault();
```
Nice update. I think the "Save Group Types" button can be deprecated since we use the form to manage the stats now, it's much more intuitive
Could you help add the "save" logic in "add new type" and "delete type"?

MichaelYuhe commented on src/popup.tsx:
```diff
@@ -60,6 +58,29 @@ const Popup = () => {
+            <button
+              className="rounded-md w-fit bg-indigo-600 px-2.5 py-1.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
+            >
+              Add New Group Type
```
```suggestion
              Add
```

MichaelYuhe reviewed (commented):
Thanks for your contribution! It's a great improvement. I've wrote 2 comments for the changes.

MichaelYuhe commented:
BTW, you can run `yarn style:fix` to lint your code before committing

flamintune commented:
### Changes Made

Removed the "Save" button and implemented direct saving functionality through "Add" or "Delete" operations. Initially, a `useEffect` hook was set up to call the save function. However, this approach was not successful. The existing codebase now directly invokes the save function upon click events, which is currently being debugged for issues.
